### PR TITLE
[refactor](wal) refactor some wal code

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/olap/wal/wal_dirs_info.cpp [hunk 1]
```diff
@@ -74,7 +71,7 @@ Status WalDirInfo::update_wal_dir_limit(size_t limit) {
         }
         size_t wal_dir_size = 0;
         RETURN_IF_ERROR(io::global_local_filesystem()->directory_size(_wal_dir, &wal_dir_size));
-        // TODO should be wal_disk_limit + wal_dir_size
+        // If here we set wal_disk_limit + wal_dir_size, then wal dir size will be larger and larger.
```
Comment (mymeiyi) [inline be/src/olap/wal/wal_dirs_info.cpp:74]: If not add, the above get wal_dir_size is unused?

**Comment by Yukang-Lian** [comment]
run buildall

**Review by mymeiyi — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**